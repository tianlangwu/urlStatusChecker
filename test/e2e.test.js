const execa = require("execa");
const {describe, test, expect} = require("@jest/globals");
expect.addSnapshotSerializer(require("jest-snapshot-serializer-ansi"));

describe("end-to-end integration", () => {
  test("prints tool with no arguments pass", async () => {
    const {stdout, stderr} = await execa("node", ["urlstatuschecker"]);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });

  test("prints tool version with arument -v", async () => {
    const {stdout, stderr} = await execa("node", ["urlstatuschecker", "-v"]);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });

  test("prints url information with a path", async () => {
    const {stdout, stderr} = await execa("node", [
      "urlstatuschecker",
      "tests/test1.html",
    ]);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });

  test("prints url information with two filepath", async () => {
    const {stdout, stderr} = await execa("node", [
      "urlstatuschecker",
      "tests/test1.html",
      "tests/test2.html",
    ]);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });

  test("prints url information with path and json output", async () => {
    const {stdout, stderr} = await execa("node", [
      "urlstatuschecker",
      "-j",
      "tests/test1.html",
    ]);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });

  test("prints url information with path and only good output", async () => {
    const {stdout, stderr} = await execa("node", [
      "urlstatuschecker",
      "-g",
      "tests/test1.html",
    ]);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });

  test("prints url information with path and only bad output", async () => {
    const {stdout, stderr} = await execa("node", [
      "urlstatuschecker",
      "-b",
      "tests/test1.html",
    ]);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });

  test("prints url information with url string", async () => {
    const {stdout, stderr} = await execa("node", [
      "urlstatuschecker",
      "-u",
      "https://www.google.com",
    ]);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });

  test("prints url information with ignore arugment", async () => {
    const {stdout, stderr} = await execa("node", [
      "urlstatuschecker",
      "-i",
      "tests/test1.html",
      "tests/test2.html",
    ]);
    expect(stdout).toMatchSnapshot();
    expect(stderr).toEqual("");
  });
});
