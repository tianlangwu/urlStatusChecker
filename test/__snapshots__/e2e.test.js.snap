// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`end-to-end integration prints tool version with arument -v 1`] = `"UrlStatusChecker version 1.0.1"`;

exports[`end-to-end integration prints tool with no arguments pass 1`] = `
"urlStatusChecker is a command line tool for check all the URL's status inside the local html file.

The command is \\"node urlStatusChecker filename\\"

Available option:
 -v / -version: Current version of urlStatusChecker

-u check URLs directly
"
`;

exports[`end-to-end integration prints url information with a path 1`] = `
"processing file \\"tests/test1.html\\"
[GOOD] https://docs.github.com/en/enterprise/2.15/user/articles/create-a-repo"
`;

exports[`end-to-end integration prints url information with ignore arugment 1`] = `
"processing ignored file tests/test2.html
processing file \\"tests/test2.html\\"
[GOOD] http://s9y.org"
`;

exports[`end-to-end integration prints url information with path and json output 1`] = `
"processing file \\"tests/test1.html\\"
{\\"url\\":\\"https://docs.github.com/en/enterprise/2.15/user/articles/create-a-repo\\",\\"status\\":\\"200\\"}"
`;

exports[`end-to-end integration prints url information with path and only bad output 1`] = `"processing file \\"tests/test1.html\\""`;

exports[`end-to-end integration prints url information with path and only good output 1`] = `
"processing file \\"tests/test1.html\\"
[GOOD] https://docs.github.com/en/enterprise/2.15/user/articles/create-a-repo"
`;

exports[`end-to-end integration prints url information with two filepath 1`] = `
"processing file \\"tests/test1.html\\"
[GOOD] https://docs.github.com/en/enterprise/2.15/user/articles/create-a-repo
processing file \\"tests/test2.html\\"
[GOOD] https://docs.github.com/en/enterprise/2.15/user/articles/create-a-repo
[GOOD] http://s9y.org"
`;

exports[`end-to-end integration prints url information with url string 1`] = `"[GOOD] https://www.google.com"`;
